commit 373b13b24006f87d3474e2c27886a5142c6bc35a
Author: nicm <nicm>
Date:   Wed May 11 20:56:58 2016 +0000

    Add on alerts rather than ignoring them if the event has already been
    queued, for example bell and activity together. From Marcel Lippmann via
    Romain Francoise.

diff --git a/alerts.c b/alerts.c
index cca0d815ef..366783511e 100644
--- a/alerts.c
+++ b/alerts.c
@@ -163,15 +163,15 @@ alerts_queue(struct window *w, int flags)
 	if (!event_initialized(&w->alerts_timer))
 		evtimer_set(&w->alerts_timer, alerts_timer, w);
 
-	if (!alerts_fired) {
+	if ((w->flags & flags) != flags) {
 		w->flags |= flags;
 		log_debug("@%u alerts flags added %#x", w->id, flags);
+	}
 
-		if (alerts_enabled(w, flags)) {
-			log_debug("alerts check queued (by @%u)", w->id);
-			event_once(-1, EV_TIMEOUT, alerts_callback, NULL, NULL);
-			alerts_fired = 1;
-		}
+	if (!alerts_fired && alerts_enabled(w, flags)) {
+		log_debug("alerts check queued (by @%u)", w->id);
+		event_once(-1, EV_TIMEOUT, alerts_callback, NULL, NULL);
+		alerts_fired = 1;
 	}
 }
 
